### PR TITLE
#50.2 get users schedule

### PR DESCRIPTION
**Pull request by Evgentik**

### CODE REVIEW

File: Recrutify/Recrutify.DataAccess/Repositories/Abstract/IScheduleRepository.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Recrutify.DataAccess.Models;
+
+namespace Recrutify.DataAccess.Repositories.Abstract
+{
+    public interface IScheduleRepository
+    {
+        Task<List<Schedule>> GetUsersSchedulesByPrimarySkillAsync(IEnumerable<Guid> interviewersIds,  DateTime date, Guid primarySkillId);
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/Abstract/IScheduleRepository.cs:10]: usersIds

File: Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs [hunk 2]
```diff
@@ -38,5 +38,14 @@ public Task IncrementCurrentApplicationsCountAsync(Guid id)
             var updateDefinition = Builders<Project>.Update.Inc(p => p.CurrentApplicationsCount, 1);
             return GetCollection().UpdateOneAsync(filter, updateDefinition);
         }
+
+        public Task<IEnumerable<Guid>> GetInterviewersIdsAsync(Guid projectid)
+        {
+            var filter = _filterBuilder.Eq(x => x.Id, projectid);
+            return GetCollection()
+             .Find(filter)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs:46]: formatting moved, should be tabs

File: Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs [hunk 3]
```diff
@@ -38,5 +38,14 @@ public Task IncrementCurrentApplicationsCountAsync(Guid id)
             var updateDefinition = Builders<Project>.Update.Inc(p => p.CurrentApplicationsCount, 1);
             return GetCollection().UpdateOneAsync(filter, updateDefinition);
         }
+
+        public Task<IEnumerable<Guid>> GetInterviewersIdsAsync(Guid projectid)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs:42]: in other methods param id, here is projectId...

Comment (elizavetapavliv) [reply Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs:42]: it's obvious that project

File: Recrutify/Recrutify.DataAccess/Repositories/Abstract/IScheduleRepository.cs (hunk 1 above)
Comment (Evgentik) [reply Recrutify/Recrutify.DataAccess/Repositories/Abstract/IScheduleRepository.cs:10]: +

File: Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+namespace Recrutify.DataAccess.Repositories
+{
+    public class ScheduleRepository : BaseRepository<Schedule>, IScheduleRepository
+    {
+        public ScheduleRepository(IOptions<MongoSettings> options)
+           : base(options)
+        {
+        }
+
+        public Task<List<Schedule>> GetUsersSchedulesByPrimarySkillAsync(IEnumerable<Guid> interviewersIds, DateTime date, Guid primarySkillId)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs:20]: usersIds

File: Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs (hunk 2 above)
Comment (Evgentik) [reply Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs:46]: +

File: Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs (hunk 3 above)
Comment (Evgentik) [reply Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs:42]: +

File: Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs (hunk 4 above)
Comment (Evgentik) [reply Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs:20]: +

File: Recrutify/Recrutify.Services/Services/ProjectService.cs [hunk 5]
```diff
@@ -114,5 +114,10 @@ public Task IncrementCurrentApplicationsCountAsync(Guid id)
         {
             return _projectRepository.IncrementCurrentApplicationsCountAsync(id);
         }
+
+        public async Task<IEnumerable<Guid>> GetInterviewersIdsAsync(Guid projectId)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Services/ProjectService.cs:118]: async await

Comment (Evgentik) [reply Recrutify/Recrutify.Services/Services/ProjectService.cs:118]: +

File: Recrutify/Recrutify.Host/Controllers/SchedulesController.cs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    public class SchedulesController : ControllerBase
+    {
+        private readonly IScheduleService _scheduleService;
+
+        public SchedulesController(IScheduleService scheduleService)
+        {
+            _scheduleService = scheduleService;
+        }
+
+        [HttpGet("users_schedules/{projectId:guid}/{primarySkillId:guid}")]
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Controllers/SchedulesController.cs:23]: remove everything and make query parameters, don't forget about [FromQuery]"users_schedules/{projectId:guid}/{primarySkillId:guid}"

File: Recrutify/Recrutify.DataAccess/Repositories/Abstract/IScheduleRepository.cs (hunk 1 above)
Comment (elizavetapavliv) [reply Recrutify/Recrutify.DataAccess/Repositories/Abstract/IScheduleRepository.cs:10]: think it's enougth to name GetByUserPrimarySkillAsync

File: Recrutify/Recrutify.Services/Services/ScheduleService.cs [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        private readonly IMapper _mapper;
+
+        public ScheduleService(IProjectService projectService, IScheduleRepository scheduleRepository, IMapper mapper)
+        {
+            _projectService = projectService;
+            _scheduleRepository = scheduleRepository;
+            _mapper = mapper;
+        }
+
+        public async Task<IEnumerable<ScheduleDTO>> GetUsersSchedulesByPrimarySkillAsync(Guid projectId, DateTime date, Guid primarySkillId)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Services/ScheduleService.cs:27]: GetByUserPrimarySkillAsync

File: Recrutify/Recrutify.Host/Controllers/SchedulesController.cs [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+        public Task<IEnumerable<ScheduleDTO>> GetUsersSchedulesByPrimarySkillAsync(Guid projectId, DateTime date, Guid primarySkillId)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Controllers/SchedulesController.cs:24]: GetByUserPrimarySkillAsync

Comment (Evgentik) [reply Recrutify/Recrutify.Host/Controllers/SchedulesController.cs:24]: +

File: Recrutify/Recrutify.Services/Services/ScheduleService.cs (hunk 7 above)
Comment (Evgentik) [reply Recrutify/Recrutify.Services/Services/ScheduleService.cs:27]: +

File: Recrutify/Recrutify.Host/Controllers/SchedulesController.cs (hunk 6 above)
Comment (Evgentik) [reply Recrutify/Recrutify.Host/Controllers/SchedulesController.cs:23]: +

File: Recrutify/Recrutify.DataAccess/Repositories/Abstract/IScheduleRepository.cs (hunk 1 above)
Comment (Evgentik) [reply Recrutify/Recrutify.DataAccess/Repositories/Abstract/IScheduleRepository.cs:10]: +

**Review by elizavetapavliv — APPROVED** [review]

**Pull request merged**